test/api_image_push_test.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"github.com/alibaba/pouch/test/environment"
+	"github.com/alibaba/pouch/test/request"
+	"github.com/go-check/check"
+)
+
+// APIImagePushSuite is the test suite for image create API.
+type APIImagePushSuite struct{}
+
+func init() {
+	check.Suite(&APIImagePushSuite{})
+}
+
+// SetUpTest does common setup in the beginning of each test.
+func (suite *APIImagePushSuite) SetUpTest(c *check.C) {
+	SkipIfFalse(c, environment.IsLinux)
+}
+
+// TestImageCreateOk tests creating an image is OK.
+func (suite *APIImagePushSuite) TestPushImageFail(c *check.C) {
+	name := "not_exist_image"
+	resp, err := request.Post("/images/" + name + "/push")
+	c.Assert(err, check.IsNil)
+	CheckRespStatus(c, resp, 500)
```

Comment:
why using 500 instead of 404 for not-exist-image?

Reply:
> 404 means no such route

I am afraid that 404 mean image not found rather than endpoint route not found.